envoy/http/filter.h

```diff
@@ -302,6 +302,21 @@ class StreamFilterCallbacks {
    * Called when filter activity indicates that the stream idle timeout should be reset.
    */
   virtual void resetIdleTimer() PURE;
+
+  /**
+   * This is a helper to get the route's per-filter config if it exists, otherwise the virtual
+   * host's. Or nullptr if none of them exist.
```

Comment:
probably document the fallback logic here.

Reply:
I am not sure. A key point to create this API is to make the searching policy transparent to the stream filters.

The filters needn't to know which policy is used for config searching.

Reply:
I think it would be better to add some more comments in the proto API of typed_per_filter_config.

Reply:
New comments should be added to the typed_per_filter_config. But it should be done in the new PR which will update the filters to use new mostSpecificPerFilterConfig.